File: 1.Two-Sum/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    
    var tempNums = [];
    for (var i = 0; i < nums.length; ++i){
        tempNums[i] = target - nums[i];
    }
    
    for (var j = 0; j < nums.length; ++j){
        for(var k = j + 1; k < tempNums.length; ++k){
            if (nums[j] === tempNums[k]){
                return [j, k];
            }else{
                continue;
            }
        }
    }
    
    return false;
};

//costs O(n^2)

File: 1.Two-Sum/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        if len(nums) <= 1:
            return False
        dic = {}
        for i in range(len(nums)):
            if nums[i] in dic:
                return [i,dic[nums[i]]]
            else:
                compliment = target - nums[i]
                dic[compliment] = i
